### PR TITLE
Bump maximum supported python to 3.12

### LINKED ISSUE #191: Support Python 3.12
Python 3.12 will be released next week, and the Python core developers [encourage package maintainers to publish updated wheels to prepare](https://www.python.org/downloads/release/python-3120rc3/). This issue requests support for the new Python release.

Blocking nvaccess/nvda#12064.

### PR DESCRIPTION
**Pull request by codeofdusk**
Closes #191.

### CODE REVIEW

File: py2exe/runtime.py [hunk 1]
```diff
@@ -3,7 +3,6 @@
 import pkg_resources
 from .dllfinder import Scanner, pydll
 
-import imp
```
Comment (codeofdusk) [inline py2exe/runtime.py:6]: Is this used anywhere? It seems not...

File: .github/workflows/matrix.json [hunk 2]
```diff
@@ -3,8 +3,10 @@
     {"python-version": "3.9", "python-arch": "x86", "run-tests": "false"},
     {"python-version": "3.10", "python-arch": "x86", "run-tests": "false"},
     {"python-version": "3.11", "python-arch": "x86", "run-tests": "false"},
+    {"python-version": "3.12", "python-arch": "x86", "run-tests": "false"},
     {"python-version": "3.8", "python-arch": "x64", "run-tests": "true"},
     {"python-version": "3.9", "python-arch": "x64", "run-tests": "true"},
     {"python-version": "3.10", "python-arch": "x64", "run-tests": "true"},
     {"python-version": "3.11", "python-arch": "x64", "run-tests": "true"},
+    {"python-version": "3.12rc3", "python-arch": "x64", "run-tests": "true"},
```
Comment (albertosottile) [inline .github/workflows/matrix.json:11]: Can I ask why you did not use "3.12" for arch x64 as well?

**Comment by albertosottile** [comment]
@codeofdusk I appreciate this contribution, but I fear enabling the build on 3.12 will only be the beginning; I expect some fixes needed before the build would be usable. Therefore, I would merge this in the `develop` branch and continue fixing the upcoming issues there.

File: setup.py [hunk 3]
```diff
@@ -173,7 +173,7 @@ def _is_debug_build():
               "Programming Language :: Python :: 3.8",
               "Programming Language :: Python :: 3.9",
               "Programming Language :: Python :: 3.10",
-              "Programming Language :: Python :: 3.11",
```
Comment (albertosottile) [inline setup.py:176]: Please do not delete the classifier for Python 3.11

File: setup_zipextimporter.py [hunk 4]
```diff
@@ -86,7 +86,7 @@ def _is_debug_build():
           setup_requires=["wheel", "cachetools", "pefile"],
           install_requires=["cachetools", "pefile"],
           platforms="Windows",
-          python_requires='>=3.7, <3.12',
```
Comment (albertosottile) [inline setup_zipextimporter.py:89]: Please do not delete the classifier for Python 3.11

File: setup.py [hunk 5]
```diff
@@ -159,7 +159,7 @@ def _is_debug_build():
           setup_requires=["wheel", "cachetools", "pefile"],
           install_requires=["cachetools", "pefile"],
           platforms="Windows",
-          python_requires='>=3.7, <3.12',
+          python_requires='>=3.7, <3.13',
```
Comment (albertosottile) [inline setup.py:162]: To solve a conflict with latest `master`, please bump the minimum version to 3.8

File: setup_zipextimporter.py [hunk 6]
```diff
@@ -86,7 +86,7 @@ def _is_debug_build():
...
+          python_requires='>=3.7, <3.13',
```
Comment (albertosottile) [inline setup_zipextimporter.py:89]: To solve a conflict with latest `master`, please bump the minimum version to 3.8

**Comment by albertosottile** [comment]
@codeofdusk Thank you for this PR, I am merging it now. Unfortunately, as I suspected, true support for 3.12 will take a while.

**Pull request merged**